mantidproject/mantid
# Fix plt.show() Not Showing the Most Recent Figure

**Description of work.**
`plt.show()` was not working due to the global figure manager not being replaced early enough. This PR refactors a lot of the startup code out of `mainwindow.py` and into a `main.py` module. The code to switch the matplotlib global figure manager out for our own is now the first thing to be run on startup. 

**To test:**
1. Ensure the Matplotlib version you are using is >= 3.1.2. Windows should have this by default, Linux users can use a virtual python environment. 
 a. importing `matplotlib` and running `matplotlib.__version__` in the python IPython console will show you. 

1. Launch mantidworkbech
2. As a script or in the IPython console run:
 a. plt.plot([1,2,3])
 b. plt.show()
3. A figure should appear with a linear line on it. 

Fixes #27374 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
